src/lib/routes/admin-api/constraints.ts

```diff
@@ -0,0 +1,35 @@
+import { Request, Response } from 'express';
+import FeatureToggleService from '../../services/feature-toggle-service';
+import { IUnleashConfig } from '../../types/option';
+import { IUnleashServices } from '../../types';
+import { IConstraint } from '../../types/model';
+import { NONE } from '../../types/permissions';
+import Controller from '../controller';
+import { Logger } from '../../logger';
+
+export default class ConstraintController extends Controller {
+    private featureService: FeatureToggleService;
+
+    private readonly logger: Logger;
+
+    constructor(
+        config: IUnleashConfig,
+        {
+            featureToggleServiceV2,
+        }: Pick<IUnleashServices, 'featureToggleServiceV2'>,
+    ) {
+        super(config);
+        this.featureService = featureToggleServiceV2;
+        this.logger = config.getLogger('/admin-api/validation.ts');
+
+        this.post('/validate', this.validateConstraint, NONE);
+    }
+
+    async validateConstraint(
```

Comment:
Wasn't this previously implemented in another controller? If it's changed, have we updated the FE counterpart?

Reply:
Right. We needed this endpoint for the segments page, where we don't have a projectId or featureName for the path. 

The frontend path is updated in the segments branch.